### ISSUE TITLE
Remember deployment details with diagram

### ISSUE THREAD

**Issue by nikku** [camunda/camunda-modeler#1066]
__Is your feature request related to a problem? Please describe.__

The deployment feature is meant to be used repeatedly (you are likely to update assets once published). Unfortunately as a user I do need to type the details (deployment name, tenant id) over and over again. 

__Describe the solution you'd like__

* [ ] The modeler should remember the entered details
* [ ] Re-publishing is as simple as opening the dialog and pressing [ENTER] or the deploy button

__Describe alternatives you've considered__

An interesting question is in which scope the details should be remembered:

* per diagram (persisted in XML) 
* per tab (stored transiently as long as the diagram tab is opened)
* in the _modeler workspace_ or _to be invented project scope_

__Solution Sketch__

If we go for the diagram.file.path prefix in the global configuration approach:

API to retrieve information per diagram:

```
config.setForDiagram(diagram, key, value);
config.getForDiagram(diagram, key, defaultValue);
```

__Considerations__

* Save As should update per diagram config entries.

---

Related to https://github.com/camunda/camunda-modeler/issues/1067, child of https://github.com/camunda/camunda-modeler/issues/1439

Related to [Support Issue](https://app.camunda.com/jira/browse/SUPPORT-5416)

**Comment by pinussilvestrus** [comment]
+1 for per tab, since different diagrams can be different endpoints referred to. 

However what do you mean with persisted in XML. Would this include storing the configuration details inside the _xml file_?

One thing makes me thinking of is the scenario when the user wants to deploy different diagrams to same endpoint. So in this case the user has to add the details for each tab. Maybe there are another options like 

- storing the last recently used configuration?
- storing configurations as templates (so I can choose previously used configuration over and over again)?

In this cases we have to think about a proper way storing data, like in https://github.com/camunda/camunda-modeler/issues/1067.

**Comment by nikku** [comment]
> However what do you mean with persisted in XML. Would this include storing the configuration details inside the _xml file_?

Yes, this is one of the options. The general question is: For how long should we remember the deployment configuration? As long as the diagram exists? As long as the diagram is open? The intuitive thing would be that the last configuration sticks to the diagram.

**Comment by pinussilvestrus** [comment]
Closed via fb0c693824c872f4b8ee75b246ff92d2b8bf5deb
